nixos/tests/systemd-initrd-vconsole.nix

```
@@ -20,10 +20,10 @@ import ./make-test-python.nix ({ lib, pkgs, ... }: {
     machine.start()
     machine.wait_for_console_text("Press Enter for maintenance")
     machine.send_console("\n")
-    machine.wait_for_console_text("Logging in with home")
+    machine.wait_for_console_text("sh-${pkgs.bash.version}#")
 
     # Check keymap
-    machine.send_console("(printf '%s to receive text: \\n' Ready && read text && echo \"$text\") </dev/tty1\n")
+    machine.send_console("(sleep 2 && printf '%s to receive text: \\n' Ready && read text && echo \"$text\") </dev/tty1\n")
     machine.wait_for_console_text("Ready to receive text:")
```

Maybe a bit more robust?

```suggestion
    while True:
      machine.send_console("echo Ready to receive commands:\n")
      try:
         machine.wait_for_console_text("Ready to receive commands:", timeout=0)
      except:
         pass
      import time; time.sleep(0.1)
 
    machine.send_console("(printf '%s to receive text: \\n' Ready && read text && echo \"$text\") </dev/tty1\n")
    machine.wait_for_console_text("Ready to receive text:")
```

---

It turns out `timeout=` does not work at all with `wait_for_console_text` due to a bug. I guess I'll add a commit to fix that to this PR...

I'm trying that though to see if it's any more robust. I think I'm gonna have to disable kernel messages too so that they aren't interleaved with the console shenanigans and cause text to be missed because it got spliced in the middle

---

I took a similar approach to what you had @Mic92 and it let me get rid of the `sh-${pkgs.bash.version}#` check. How's the most recent push look?

Also cc @RaitoBezarius @alyssais from #234513:

I've added a commit to fix the `timeout=` option for `wait_for_console_text` which seemed to not work with `retry` (since it passes in a `last_try` positional argument which `console_matches` does not accept)

---

Note that for this to be backported (so that the test stops wasting time in the release-23.05 jobset), then either #234513 will have to be backported as well or a different solution for this PR that does not rely on that will need to be backported (I'm indifferent since it's a non-end-user-facing change and not breaking afaict)

---

Fixing the timeout also in 23.05 is worth it. It's not just nixos itself that uses it for testing.

---

Right now 23.05 does not support `timeout=` on `wait_for_console_text` at all (it was added post-branch-off), so backporting the other PR wouldn't really "fix" anything aside from this one test. But it's also backwards compatible so if @RaitoBezarius is good with it being backported, then I'm good with it being backported

I'll go apply the tag and once the other PR is backported, I'll re-run the backport on this PR since it should then work

---

Oops I just realized it would need the follow-up PR #234607 too for backport. This probably would have been better as a manual backport...

I'll just push the follow-up to the backport branch for the other PR too (sorry for making a mess -- clicked buttons to add labels before I thought too much about it)